src/StreamJsonRpc/Reflection/MethodSignatureAndTarget.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace StreamJsonRpc
+{
+    using System;
+    using System.Reflection;
+    using Microsoft;
+
+    internal struct MethodSignatureAndTarget : IEquatable<MethodSignatureAndTarget>
+    {
+        public MethodSignatureAndTarget(MethodInfo method, object target)
+        {
+            Requires.NotNull(method, nameof(method));
+
+            this.Signature = new MethodSignature(method);
+            this.Target = target;
+        }
+
+        public MethodSignature Signature { get; }
+
+        public object Target { get; }
+
+        public override bool Equals(object obj)
+        {
+            return obj is MethodSignatureAndTarget other
+                && this.Equals(other);
+        }
+
+        public bool Equals(MethodSignatureAndTarget other)
+        {
+            return this.Signature.Equals(other.Signature)
+                && object.Equals(this.Target, other.Target);
```

Comment:
Should we use ReferenceEquals here? We probably don't want to call Equals on target objects.

Reply:
Although we have test coverage here, there is no way the product (at present) will actually find a different Target object in this method (since we never invoke it with a different one). I've made the correction you suggest since I agree it's accurate. It's just unfortunate we can't test it. The alternative would be to throw in the currently untestable case, but since we are pretty confident about what the correct behavior is, and it's cheap, I'm going with your recommendation.